### PR TITLE
APS-1362 Add documentation/configuration to run e2e tests against dev upstream

### PR DESCRIPTION
**Pull request by davidatkinsuk**
This PR adds documentation for how to run the e2e tests using local ui/api, but upstream dev services. It also simplifies the playwright config and adds a new project to support running against upstream dev services.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,6 +181,54 @@ notify:
...
+HMPPS_AUTH_USERNAME=AP_USER_TEST_1
+SYSTEM_CLIENT_ID=approved-premises-api
+ENABLE_V2_MATCH=true
+```
+
+Then start using one of
+
+```
+npx playwright test --config ./e2e/playwright.config.ts  --project=local-dev-upstream
+npx playwright test --config ./e2e/playwright.config.ts  --project=local-dev-upstream --ui
```
Comment (froddd) [inline README.md:229]: These could also be added to the `package.json`. If the env vars above are in `.env.dev-upstream`, then the following scripts could be added to it:

```
"test:e2e:dev-upstream": "export $(cat feature.env) && npx playwright test --config ./e2e/playwright.config.ts  --project=local-dev-upstream",
"test:e2e:dev-upstream:ui": "npm run test:e2e:local:dev-upstream -- --ui"
```

The above README section would then list `npm run test:e2e:dev-upstream` and `test:e2e:dev-upstream:ui` to start using.

File: e2e/playwright.config.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { TestOptions } from '@approved-premises/e2e'
 
 export default defineConfig<TestOptions>({
   testDir: './',
+  testMatch: /.*\.*(setup)|(spec)\.ts/,
```
Comment (froddd) [inline e2e/playwright.config.ts:7]: I think the setup steps need to remain as they are, as they are necessary to ensure the user is first logged out before running any test -- these special steps run _before_ any other feature is started.

File: README.md [hunk 3]
```diff
@@ -181,6 +181,54 @@ notify:
...
+CAS1_E2E_USER_WITHOUT_ROLES_PASSWORD=doesntmatter
+CAS1_E2E_USER_WITHOUT_ROLES_USERNAME=AP_USER_TEST_1
+CAS1_E2E_CHECK_EMAILS=false
+HMPPS_AUTH_EXTERNAL_URL=http://localhost:9091/auth
+HMPPS_AUTH_NAME=AP_USER TEST_1
+HMPPS_AUTH_PASSWORD=*REDACTED*
+HMPPS_AUTH_URL=http://localhost:9091/auth
+HMPPS_AUTH_USERNAME=AP_USER_TEST_1
+SYSTEM_CLIENT_ID=approved-premises-api
+ENABLE_V2_MATCH=true
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply README.md:222]: Moving all this to .env file is my plan for the next PR (if i get there)

The passwords don't actually matter in this case because we're still using our local hmpps-auth for user auth/JWT validation

[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply README.md]: I've pushed a new commit moving all the sample config to .env files and simplifying the README. I think we need to decide which of these 3 ways of running we actually need going forward (once we've harmonised the environment config maybe)

File: README.md [hunk 4]
```diff
@@ -181,6 +181,54 @@ notify:
   guest-list-api-key: aforementioned NOTIFY_API_KEY value
 ```
 
+#### Running locally but using upstream dev services (emulate tests that run in 'dev')
+
+Start ap-tools as follows:
+
+```shell
+docker compose down --clear-databases
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply README.md]: oops, correct, this is fixed

File: README.md (hunk 1 above)
Comment (davidatkinsuk) [reply README.md]: i've added these. we probably need to review the existing package entries for running e2e as they're a bit confusing

FYI i've called this 'local-dev-upstream'. it's a bit wordy but i think it's a succinct as it could be?

File: e2e/playwright.config.ts (hunk 2 above)
Comment (davidatkinsuk) [reply e2e/playwright.config.ts:7]: Let's discuss. I had assumed 'dependencies' just meant 'config to merge in' but if it's affecting the 'pre tests' than yes this needs a rethink

Comment (davidatkinsuk) [reply e2e/playwright.config.ts]: No need to discuss, i've put the setup projects back in

File: package.json [hunk 5]
```diff
@@ -28,9 +28,11 @@
     "test:integration": "npm run start-test-wiremock && start-server-and-test start-feature http://localhost:3007/ping int-test",
     "test:integration:ui": "npm run start-test-wiremock && start-server-and-test start-feature:dev http://localhost:3007/ping int-test-ui",
     "test:e2e:ci": "npx playwright test --config ./e2e/playwright.config.ts --project=dev",
-    "test:e2e:local": "npx playwright test --config ./e2e/playwright.config.ts --project=local",
     "test:e2e:ui": "npm run test:e2e:ci -- --ui",
+    "test:e2e:local": "npx playwright test --config ./e2e/playwright.config.ts --project=local",
     "test:e2e:local:ui": "npm run test:e2e:local -- --ui",
+    "test:e2e:local-dev-upstream": "npx playwright test --config ./e2e/playwright.config.ts --project=local-dev-upstream",
```
Comment (froddd) [inline package.json:34]: If there was env-specific `.env` files, this could be something along those lines:

```suggestion
    "test:e2e:local-dev-upstream": "export $(cat .env.dev) && npx playwright test --config ./e2e/playwright.config.ts --project=local-dev-upstream",
```

Comment (davidatkinsuk) [reply package.json:34]: that's great, thanks Fred. i'll have a play around with this

**Review by froddd — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (davidatkinsuk) [reply package.json:34]: as discussed, we'll merge the PR as-is and pick this up in a subsequent PR
